pytket/pytket/wasm/wasm.py

```diff
@@ -34,6 +36,107 @@ def __init__(self, filepath: str):
 
         self._wasmuid = hashlib.md5(self._wasm_file_encoded).hexdigest()
 
+        wasm_module = Module(Store(), open(self._filepath, "rb").read())
+        instance = Instance(wasm_module)
```

Comment:
The validation only works with the file, not with the created module. You can see that in the commit I added: @cqc-melf
[add validation of the module](https://github.com/CQCL/tket/pull/561/commits/51ab66b8eac47b13318d8bfff3f21f01592116b4)

I am not sure if we should keep that, because it can take quiet some time for bigger modules. That was the reason why I had not added that in the first draft.